tests/baselines/reference/genericCallWithGenericSignatureArguments.types

```diff
@@ -0,0 +1,225 @@
+=== tests/cases/conformance/types/typeRelationships/typeInference/genericCallWithGenericSignatureArguments.ts ===
+// When a function expression is inferentially typed (section 4.9.3) and a type assigned to a parameter in that expression references type parameters for which inferences are being made, 
+// the corresponding inferred type arguments to become fixed and no further candidate inferences are made for them.
+
+function foo<T>(a: (x: T) => T, b: (x: T) => T) {
+>foo : <T>(a: (x: T) => T, b: (x: T) => T) => (x: T) => T
+>T : T
+>a : (x: T) => T
+>x : T
+>T : T
+>T : T
+>b : (x: T) => T
+>x : T
+>T : T
+>T : T
+
+    var r: (x: T) => T;
+>r : (x: T) => T
+>x : T
+>T : T
+>T : T
+
+    return r;
+>r : (x: T) => T
+}
+
+//var r1 = foo((x: number) => 1, (x: string) => ''); // error
+var r1b = foo((x) => 1, (x) => ''); // {} => {}
+>r1b : (x: {}) => {}
+>foo((x) => 1, (x) => '') : (x: {}) => {}
+>foo : <T>(a: (x: T) => T, b: (x: T) => T) => (x: T) => T
+>(x) => 1 : (x: {}) => number
+>x : {}
+>1 : 1
+>(x) => '' : (x: {}) => string
+>x : {}
+>'' : ""
+
+var r2 = foo((x: Object) => null, (x: string) => ''); // Object => Object
+>r2 : (x: any) => any
+>foo((x: Object) => null, (x: string) => '') : (x: any) => any
+>foo : <T>(a: (x: T) => T, b: (x: T) => T) => (x: T) => T
+>(x: Object) => null : (x: Object) => any
+>x : Object
+>Object : Object
+>null : null
+>(x: string) => '' : (x: string) => string
+>x : string
+>'' : ""
+
+var r3 = foo((x: number) => 1, (x: Object) => null); // number => number
+>r3 : (x: any) => any
+>foo((x: number) => 1, (x: Object) => null) : (x: any) => any
+>foo : <T>(a: (x: T) => T, b: (x: T) => T) => (x: T) => T
+>(x: number) => 1 : (x: number) => number
+>x : number
+>1 : 1
+>(x: Object) => null : (x: Object) => any
+>x : Object
+>Object : Object
+>null : null
+
+var r3ii = foo((x: number) => 1, (x: number) => 1); // number => number
+>r3ii : (x: number) => number
+>foo((x: number) => 1, (x: number) => 1) : (x: number) => number
+>foo : <T>(a: (x: T) => T, b: (x: T) => T) => (x: T) => T
+>(x: number) => 1 : (x: number) => number
+>x : number
+>1 : 1
+>(x: number) => 1 : (x: number) => number
+>x : number
+>1 : 1
+
+var a: { x: number; y?: number; };
+>a : { x: number; y?: number; }
+>x : number
+>y : number
+
+var b: { x: number; z?: number; };
+>b : { x: number; z?: number; }
+>x : number
+>z : number
+
+var r4 = foo((x: typeof a) => a, (x: typeof b) => b); // typeof a => typeof a
+>r4 : (x: { x: number; y?: number; }) => { x: number; y?: number; }
+>foo((x: typeof a) => a, (x: typeof b) => b) : (x: { x: number; y?: number; }) => { x: number; y?: number; }

```

Comment:
Previously we'd require that one be an exact subtype of the other, which would fail. Now, we allow the call because assignability succeeds even though neither is a subtype of the other. I don't think this corner case much matters, and you can make just as good a case for succeeding as for failing.